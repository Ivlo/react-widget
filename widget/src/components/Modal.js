import React from "react";

const Modal = ({ children, onClickClose }) => {
  return (
    <div className="modal">
      <div className="modal__content">{children}</div>
      <span href="" onClick={onClickClose} className="modal__action">
        X
      </span>
    </div>
  );
};

export default Modal;
