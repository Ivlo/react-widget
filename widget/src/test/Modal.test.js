import React from "react";
import { shallow } from "enzyme";
import Modal from "../components/Modal";

const callbackClose = jest.fn();
const children = "Content Modal";

describe("Modal", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Modal onClickClose={callbackClose}>{children}</Modal>);
  });

  it("should render the Modal Component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("the content of our Modalcontent element should be our children prop", () => {
    const modalContent = wrapper.find(".modal__content");
    expect(modalContent.text()).toEqual(children);
  });

  it("should be render with a close element", () => {
    const modalAction = wrapper.find(".modal__action");
    expect(modalAction.length).toBe(1);
  });

  it("close action callback should be execute when click on modal__action element", () => {
    const modalAction = wrapper.find(".modal__action");
    modalAction.simulate("click");
    expect(callbackClose).toHaveBeenCalled();
  });
});
