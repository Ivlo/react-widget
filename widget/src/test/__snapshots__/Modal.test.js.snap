// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render the Modal Component correctly 1`] = `
<div
  className="modal"
>
  <div
    className="modal__content"
  >
    Content Modal
  </div>
  <span
    className="modal__action"
    href=""
    onClick={[MockFunction]}
  >
    X
  </span>
</div>
`;
